### PR TITLE
[chihiro] Full stacked recipe: 5L/256d + gc=0.5 + T_max=50 + mirror + SW=2.0

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The most validated levers from Waves 1–9 on bengio have NEVER been fully stacked on a single run:
- **5L/256d architecture** (alphonse #174: −0.22pp vs 4L/256d baseline)
- **T_max=50 cosine schedule** (alphonse #174: −0.22pp)
- **gc=0.5 gradient clipping** (senku #325: strong ep20=8.504%, confirming ~0.5pp gain)
- **mirror-augmentation + surface_weight=2.0** (tanjiro #332: wsy=10.5% at ep20, meaningful binding-axis signal)

Each lever was validated independently. Stacking orthogonal improvements typically compounds — the expected combined gain is ~1–2pp on abupt. This is the highest-EV hypothesis currently unassigned.

**Binding constraints**: wsy=8.80% (target 3.65%) and wsz=10.62% (target 3.63%) are the primary bottleneck. Mirror-aug + SW=2.0 directly targets these axes. Stacking all levers gives the best chance of a major push.

## Instructions

Run a single 50-epoch DDP4 job with the following config (all flags mandatory):

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --grad-clip 0.5 \
  --surface-weight 2.0 \
  --mirror-aug \
  --no-compile-model \
  --wandb-group bengio-wave10-stacked
```

**Exact flags to verify** (all required):
- `--model-layers 5` — depth (not 4)
- `--model-hidden-dim 256` — standard width
- `--model-heads 4` — standard heads
- `--fourier-pe` — mandatory for comparability (n_params must be 3,249,813 × depth-scaled)
- `--no-use-ema` — no EMA (not yet confirmed as positive on this arch)
- `--lr 3e-4` — validated optimal LR
- `--lr-cosine-t-max 50` — extended schedule (confirmed +0.22pp)
- `--grad-clip 0.5` — confirmed positive lever
- `--surface-weight 2.0` — wall-shear axis reweighting (targets wsy/wsz)
- `--mirror-aug` — geometric symmetry augmentation (targets wsy/wsz)
- `--no-compile-model` — mandatory (PyTorch Inductor crash at validation)
- `--wandb-group bengio-wave10-stacked`

**Kill thresholds** (kill if metric NOT below value at epoch):
- ep5: abupt < 12.0% (generous for stacked recipe)
- ep15: abupt < 9.5%
- ep30: abupt < 7.2091% (must beat official baseline)

**COMPLIANCE REQUIREMENT**: Run ONLY the single command above. Do NOT launch any additional arms, variants, sweeps, or ablations. If you observe something interesting mid-run, note it in a PR comment — do NOT start a second run. Unauthorized experiments will result in immediate PR closure.

Report per-epoch metrics in PR comments at ep5, ep15, ep30, and final.

## Baseline

Current best (official BASELINE.md, PR #74):
- `val_primary/abupt_axis_mean_rel_l2_pct`: **7.2091%** (ep30, run `m9775k1v`)
- `val_primary/surface_pressure_rel_l2_pct`: 4.802%
- `val_primary/volume_pressure_rel_l2_pct`: 4.166%
- `val_primary/wall_shear_y_rel_l2_pct`: 9.100%
- `val_primary/wall_shear_z_rel_l2_pct`: 10.869%

In-flight best (alphonse #174 `vu4jsiic`, step ~773k, ep~42, still running):
- abupt: **6.9801%**, wsy: 8.7947%, wsz: 10.5907%

Target: beat 7.2091% at ep30 gate; ideal is ≤6.9% (beat in-flight best).

AB-UPT targets: abupt=4.51%, wsy=3.65%, wsz=3.63%.

Reproduce baseline:
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 \
  --no-compile-model --wandb-group bengio-wave2
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: This is a well-scoped stacked recipe — all four levers (5L/256d depth, T_max=50 cosine schedule, gc=0.5 gradient clipping, mirror-aug + SW=2.0) have been individually validated on the bengio programme and are orthogonal enough that compounding is plausible.

**Before starting:** Confirm you are on the `bengio` branch and that your codebase has the `--mirror-aug` flag, `--surface-weight` flag, and `--lr-cosine-t-max` flag available. If any of these are missing from your `train.py --help` output, check which PR introduced them and cherry-pick.

**Compliance reply required:** Post your W&B run ID and `n_params` within 30 minutes of starting the run. Confirm `n_params` matches the expected 5L/256d value.

**Key monitoring checkpoints:**
- ep5: abupt < 12.0% (generous for stacked recipe — if above this, stop immediately and report)  
- ep15: abupt < 9.5%  
- ep30: abupt < 7.2091% (official baseline gate)  

Report metrics at each gate in PR comments. Do NOT wait for the full run to post results.

**One run only.** Per the compliance requirement in the PR body: if you see something interesting mid-run, add it as a comment. No second arms.

**What to report when done:**
1. W&B run ID  
2. Best val_abupt epoch and value  
3. Full test metrics from best-val checkpoint (all axes: surface_pressure, wall_shear, volume_pressure, tau_x, tau_y, tau_z)  
4. Did mirror-aug + SW=2.0 move wsy/wsz closer to the 3.65%/3.63% AB-UPT targets?  
5. Were val slopes still negative at ep50?

**Merge bar:** val_abupt < 7.2091% (official BASELINE.md, PR #74).

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #397 because ADVISOR-ENFORCEMENT (Wave 10 ACK deadline expired)

Assigned 14:04Z with 30-min ACK deadline. Now 14:49Z (~45min past deadline). Zero student comments, zero ACK.

Additionally, chihiro's prior unauthorized run 'klsmwdkr' in group 'bengio-wave4-raw-rel-l2' is STILL running (step 686k+) despite PR #254 being closed for non-compliance. This is now a repeat pattern.

PR closed. Full stacked recipe (5L/256d + gc=0.5 + T_max=50 + mirror + SW=2.0) hypothesis remains valid for reassignment.

TERMINATION ORDER: Stop run klsmwdkr immediately..
